### PR TITLE
common component text field

### PR DESCRIPTION
**Pull request by ahmadysalah**
create two textfield common component
1- first for text failed can user write inside and required
2- second text field is disabled input
3- create css file and import it in both components

### CODE REVIEW

File: client/src/components/common/TextField/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  type: '',
+  label: '',
+  value: '',
+};
+
+InputText.propTypes = {
+  id: PropTypes.string,
+  type: PropTypes.string,
+  value: PropTypes.string,
+  label: PropTypes.string,
```
Comment (Asmaathabet) [inline client/src/components/common/TextField/index.js:33]: complete adding type for the rest of props according my last edits.

File: client/src/components/common/TextField/textdisabled.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    <div className={classes.root}>
+      <TextField
+        disabled
+        id="outlined-disabled"
+        label={label}
+        defaultValue={value}
+        className={classes.container}
+        variant="outlined"
+      />
+    </div>
```
Comment (Asmaathabet) [inline client/src/components/common/TextField/textdisabled.js:18]: there is no need for this file, cause we can use on component for text filed and add disabled property as props for it when needed , just delete it & change what is needed append on that .

File: client/src/components/common/TextField/index.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      />
+    </div>
+  );
+}
+
+InputText.propTypes = {
+  id: PropTypes.string,
+  type: PropTypes.string,
+  label: PropTypes.string,
+  value: PropTypes.string,
```
Comment (Asmaathabet) [inline client/src/components/common/TextField/index.js:41]:
```suggestion
  value: PropTypes.string.isRequired,
```

File: client/src/components/common/TextField/index.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  className: PropTypes.string,
+  message: PropTypes.string,
+  placeholder: PropTypes.string,
```
Comment (Asmaathabet) [inline client/src/components/common/TextField/index.js:44]:
```suggestion
  placeholder: PropTypes.string.isRequired,
```

File: client/src/components/common/TextField/index.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        disabled={disabled}
+        variant="outlined"
+      />
+    </div>
+  );
+}
+
+InputText.propTypes = {
+  id: PropTypes.string,
+  type: PropTypes.string,
```
Comment (Asmaathabet) [inline client/src/components/common/TextField/index.js:39]:
```suggestion
  type: PropTypes.string.isRequired,
```

File: client/src/components/common/TextField/index.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  label: PropTypes.string,
+  value: PropTypes.string,
+  className: PropTypes.string,
+  message: PropTypes.string,
+  placeholder: PropTypes.string,
+  onChange: PropTypes.func,
```
Comment (Asmaathabet) [inline client/src/components/common/TextField/index.js:45]:
```suggestion
  onChange: PropTypes.func.isRequired,
```

File: client/src/components/common/TextField/index.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  id: 'outlined-required',
+  type: '',
+  label: '',
+  value: '',
+  className: '',
+  message: '',
+  placeholder: '',
+  onChange: '',
+  disabled: false,
+};
```
Comment (Asmaathabet) [inline client/src/components/common/TextField/index.js:58]: remove required props from here, I mean type, value, placeholder, onChange.

File: client/src/components/common/TextField/index.js (hunk 5 above)
Comment (ahmadysalah) [reply client/src/components/common/TextField/index.js:39]: i think type is not Required

File: client/src/components/common/TextField/index.js [hunk 8]
```diff
@@ -38,22 +38,19 @@ InputText.propTypes = {
...
   message: PropTypes.string,
-  placeholder: PropTypes.string,
-  onChange: PropTypes.func,
+  placeholder: PropTypes.string.isRequired,
+  onChange: PropTypes.func.isRequired,
   disabled: PropTypes.bool,
 };
 InputText.defaultProps = {
   id: 'outlined-required',
   type: '',
```
Comment (Asmaathabet) [inline client/src/components/common/TextField/index.js:50]: add default type for text field.
```suggestion
  type: 'text',
```

**Pull request merged**